package.json

```diff
@@ -12,11 +12,17 @@
   ],
   "scripts": {
     "build": "tsc",
-    "test": "NODE_ENV=test nyc mocha ${MOCHA_OPTS}",
+    "pretest": "npm run build",
```

Comment:
I don't think all of these changes are right. It's not clear that testing will actually always build the code, and some of the scripts seem duplicative or empty.

Reply:
`pretest` is magical ... and with what was here before `test` _never_ ran build automatically.

This layout is basically what we have in our other repos -- `test-ci` _just_ runs the tests, whereas `test` is for humans and does a build-then-test

Reply:
Also, FWIW, `pre` is the magic bit -- you can add a `pre<foo>` target for just about anything (and a `post<foo>` one too) and `npm` will know to run them at the appropriate time.

`rush` is not so smart, but that's not an issue here.

Reply:
I think you were looking at the versions here before my morning commit which cleaned stuff up. For one thing, precommit did all-the-things, which was desirable.

Reply:
`precommit` still does all-the-things now -- it runs `lint` then `test`.  `test` runs `build` and then `test-ci`, which actually runs the mocha stuff.

Note that I updated the CI config as part of this so it won't double-build -- it runs `build`, then `test-ci`